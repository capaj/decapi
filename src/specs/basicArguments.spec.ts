import {
  Arg,
  compileSchema,
  Context,
  Field,
  ObjectType,
  Query,
  SchemaRoot
} from '../../src/index.js'

import { printSchema } from 'graphql'

describe('arguments', () => {
  it('should compile args correctly', () => {
    @ObjectType()
    class Hello {
      @Field()
      argRename(@Arg({ name: 'myFancyArgument' }) a: string): boolean {
        expect(a).toBe('fancy')

        // @ts-expect-error 3/21/2022
        return null
      }

      @Field()
      argRename2(
        @Arg({ name: 'myFirstFancyArgument' }) a: string,
        @Arg({ name: 'mySecondFancyArgument' }) b: string | null,
        @Context ctx: any
      ): boolean {
        expect(ctx).toBe(null)
        expect(a).toBe('fancya')
        expect(b).toBe('fancyb')

        return false
      }
    }
    @SchemaRoot()
    class BarSchema {
      @Query()
      hello(): Hello {
        return new Hello()
      }
    }

    const schema = compileSchema(BarSchema)

    expect(printSchema(schema)).toMatchSnapshot()
  })
})

describe('onlyDecoratedArgs', function () {
  it('should omit args when onlyDecoratedArgs is used', async () => {
    class CustomClass {}

    @ObjectType()
    class Test {
      @Field({ onlyDecoratedArgs: true })
      shouldHaveNone(a1?: Function, a2?: CustomClass): boolean {
        return false
      }

      @Field({ onlyDecoratedArgs: true })
      shouldHaveOnlyOne(v1: boolean, @Arg() v2: boolean): boolean {
        return v2
      }
    }

    @SchemaRoot()
    class FooSchema {
      @Query()
      test(): Test {
        return new Test()
      }
    }

    const schema = compileSchema(FooSchema)
    expect(printSchema(schema)).toMatchSnapshot()
  })
})
