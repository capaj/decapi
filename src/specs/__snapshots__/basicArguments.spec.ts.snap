// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arguments should compile args correctly 1`] = `
"type Query {
  hello: Hello
}

type Hello {
  argRename(myFancyArgument: String!): Boolean!
  argRename2(myFirstFancyArgument: String!, mySecondFancyArgument: String): Boolean!
}"
`;

exports[`onlyDecoratedArgs should omit args when onlyDecoratedArgs is used 1`] = `
"type Query {
  test: Test
}

type Test {
  shouldHaveNone: Boolean!
  shouldHaveOnlyOne(v2: Boolean!): Boolean!
}"
`;
